### ISSUE TITLE
rdiv! on LU object is much slower than ldiv!

### ISSUE THREAD

**Issue by stevengj** [JuliaLang/julia#55422]
From [this post](https://discourse.julialang.org/t/fastest-way-to-perform-a-a-b-a-b-where-a-and-b-are-vectors-of-indices/117958/2?u=stevengj), I'm finding that `rdiv!` on LU objects (added in #31285) is much slower than `ldiv!`.

For example:
```jl
using LinearAlgebra, BenchmarkTools
function inv2(A)
    LU = lu(A)
    return ldiv!(LU, Matrix{eltype(LU)}(I, size(A)...))
end
function inv3(A)
    LU = lu(A)
    return rdiv!(Matrix{eltype(LU)}(I, size(A)...), LU)
end
```
gives
```
julia> A = rand(1000,1000); @btime inv($A); @btime inv2($A); @btime inv3($A);
  10.791 ms (5 allocations: 8.13 MiB)
  10.796 ms (5 allocations: 15.27 MiB)
  912.368 ms (5 allocations: 15.27 MiB)

julia> inv(A) ≈ inv2(A) ≈ inv3(A)
true
```

**Comment by stevengj** [comment]
I benchmarked `rdiv!` and `ldiv!` on `UpperTriangular` and `UnitLowerTriangular` matrices (which *should* represent most of the time for an LU solve), and they are comparable (both taking a few seconds for 1000x1000 matrices), so it's not obvious to me what the problem could be.  What's taking hundreds of ms here?

**Comment by KristofferC** [comment]
All time seem to be spent in

https://github.com/JuliaLang/julia/blob/4954197196d657d14edd3e9c61ac101866e6fa25/stdlib/LinearAlgebra/src/triangular.jl#L1162

Since it is an adjoint it is iterating there in a cache unfriendly way (row by row). Not sure if that is the full answer.

**Comment by stevengj** [comment]
Why would it be calling `generic_trimatdiv!` instead of a BLAS call?

**Comment by KristofferC** [comment]
It dispatches to

https://github.com/JuliaLang/julia/blob/3db1d19c91cc922734e476ce75a35e1f048dae86/stdlib/LinearAlgebra/src/triangular.jl#L1014-L1018

**Comment by aravindh-krishnamoorthy** [comment]
> I benchmarked `rdiv!` and `ldiv!` on `UpperTriangular` and `UnitLowerTriangular` matrices (which _should_ represent most of the time for an LU solve), and they are comparable (both taking a few seconds for 1000x1000 matrices), so it's not obvious to me what the problem could be. What's taking hundreds of ms here?

The problem is that LAPACK's `getrs!(trans,A,ipiv,B)` can only apply transposition (`trans`) to `A`, whereas https://github.com/JuliaLang/julia/blob/1eabe90fa054abc74f541798ae4dfcc9445bb564/stdlib/LinearAlgebra/src/lu.jl#L789 applies transposition to both `A` and `B`.

With
```julia
julia> LinearAlgebra.rdiv!(B::AbstractMatrix, A::LU) = transpose(ldiv!(transpose(A), copy(transpose(B))))
```
the performance is more reasonable:
```julia
julia> A = rand(1000,1000); @btime inv($A); @btime inv2($A); @btime inv3($A);
  18.078 ms (5 allocations: 8.13 MiB)
  24.587 ms (5 allocations: 15.27 MiB)
  26.090 ms (7 allocations: 22.90 MiB)
```

However, this requires a copy (mainly memory impact):
```julia
julia> @btime copy(A);
  242.988 μs (2 allocations: 7.63 MiB)
```

**Comment by aravindh-krishnamoorthy** [comment]
**Update:** As indicated by @dkarrasch in the comments of the now-closed PR #55760, this is **not** a missing functionality issue, but a dispatch issue. See #55764 for a possible solution.

**Comment by dkarrasch** [comment]
Indeed. Unfortunately, I have very little time at the moment, so if somebody could debug the "lu of tridiagonal" case in that PR (everything else is working just fine), that would be awesome!
